java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspTemplateUI.java

```diff
@@ -0,0 +1,253 @@
+                        defaultTargetExact[0] = false;
+                        continue;
+                    }
+                    return DataFolder.findFolder(fo);
+                }
+            } catch (MalformedURLException ex) {
+                continue;
+            }
+        }
+        defaultTargetExact[0] = false;
+        return findWorkspaceRoot(folders);
+    }
+
+    private static String removeExtensionFromFileName(String nameWithExtension, String templateExtension) {
+        if (nameWithExtension.endsWith('.' + templateExtension)) {
+            return nameWithExtension.substring(0, nameWithExtension.length() - templateExtension.length() - 1);
+        } else {
+            return nameWithExtension;
+        }
+    }
+
+    private static List<QuickPickItem> quickPickTemplates(final DataFolder folder) {
+        Node[] arr = folder.getNodeDelegate().getChildren().getNodes(true);
+        List<QuickPickItem> categories = new ArrayList<>();
+        for (Node n : arr) {
+            FileObject fo = n.getLookup().lookup(FileObject.class);
+            DataObject obj = n.getLookup().lookup(DataObject.class);
+
+            boolean display;
+            if (obj instanceof DataFolder) {
+                Object o = obj.getPrimaryFile ().getAttribute ("simple"); // NOI18N
+                display = o == null || Boolean.TRUE.equals (o);
+            } else {
+                display = obj.isTemplate();
+            }
+
+            if (display) {
+                String detail = findDetail(obj);
+                final String displayName = n.getDisplayName();
+                String description = n.getShortDescription();
+                if (description != null && description.equals(displayName)) {
+                    description = null;
+                }
+                categories.add(new QuickPickItem(
+                    displayName, description, detail,
+                    false, fo.getNameExt()
+                ));
+            }
+        }
+        return categories;
+    }
+
+    private static String findDetail(DataObject obj) {
+        URL description = org.openide.loaders.TemplateWizard.getDescription(obj);
+        String descriptionText = null;
+        if (description != null) {
+            try (ByteArrayOutputStream os = new ByteArrayOutputStream(); InputStream is = description.openStream()) {
+                FileUtil.copy(is, os);
+                descriptionText = os.toString("UTF-8");
+                int bodyBegin = descriptionText.toUpperCase().indexOf("<BODY>");
+                int bodyEnd = descriptionText.toUpperCase().indexOf("</BODY>");
+                if (bodyBegin >= 0 && bodyEnd > bodyBegin) {
+                    descriptionText = descriptionText.substring(bodyBegin + 6, bodyEnd);
+                } else {
+                    descriptionText = null;
+                }
+            } catch (IOException ex) {
+                Exceptions.printStackTrace(ex);
```

Comment:
This could raise an error message in the LSP client's UI through `DialogDisplayer.notify`.

Reply:
Lowering severity in 11fb117581e18 before printing the stack.